workflow-controller/controller-service/src/main/java/io/hyscale/controller/commands/deploy/HyscaleDeployServiceCommand.java

```diff
@@ -126,53 +123,84 @@
+
+    private List<Validator<WorkflowContext>> postValidators;
+
+    @PostConstruct
+    public void init() {
+        this.postValidators = new LinkedList<>();
+        this.postValidators.add(dockerValidator);
+        this.postValidators.add(registryValidator);
+        this.postValidators.add(manifestValidator);
+        this.postValidators.add(clusterValidator);
+        this.postValidators.add(volumeValidator);
+    }
+
     @Override
     public Integer call() throws Exception {
-        WorkflowLogger.header(ControllerActivity.PROCESSING_INPUT);
         if (!CommandUtil.isInputValid(this)) {
             return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
-        HyscaleCommandSpecBuilder commandSpecBuilder = new HyscaleCommandSpecBuilder();
-        commandSpecBuilder.setAppName(appName);
-        commandSpecBuilder.setServiceSpecFiles(serviceSpecs);
-        
-        if (profileArg != null) {
-            commandSpecBuilder.setProfileFiles(profileArg.getProfiles());
-            commandSpecBuilder.setProfileName(profileArg.getProfileName());
-        }
-        // Handles input preprocessing
-        HyscaleInputSpec hyscaleInput = hyscaleInputSpecProcessor.process(commandSpecBuilder);
-        if (hyscaleInput == null) {
+
+        // Validate Service specs with schema
+        if (!serviceSpecInputValidator.validate(serviceSpecsFiles)) {
             return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
-        
+
         Map<String, File> serviceVsSpecFile = new HashMap<String, File>();
-        for (File serviceSpec : serviceSpecs) {
+        for (File serviceSpec : serviceSpecsFiles) {
             serviceVsSpecFile.put(ServiceSpecUtil.getServiceName(serviceSpec), serviceSpec);
         }
-        List<EffectiveServiceSpec> effectiveServiceSpecList = serviceSpecProcessor.getEffectiveServiceSpec(hyscaleInput);
-        
-        List<WorkflowContext> contextList = workflowContextBuilder.buildContextList(effectiveServiceSpecList, appName, namespace);
-        
-        contextList = workflowContextBuilder.updateAuthConfig(contextList);
-        
-        postValidatorsProvider.getDeployPostValidators().forEach( each -> inputSpecPostValidator.addValidator(each)); 
-        
+
+        // Process servicespecs to form EffectiveServiceSpec from ProfileArg & ServiceSpecFiles
+        List<EffectiveServiceSpec> effectiveServiceSpecs = serviceSpecProcessor.process(profileArg, serviceSpecsFiles);
+
+        // Construct WorkflowContext
+        List<WorkflowContext> contextList = effectiveServiceSpecs.stream().filter(each -> {
+            return each != null && each.getServiceSpec() != null && each.getServiceMetadata() != null;
+        }).map(each -> {
+            WorkflowContextBuilder builder = new WorkflowContextBuilder(appName);
+            try {
+                builder.withProfile(each.getServiceMetadata().getEnvName());
+                builder.withAuthConfig(authConfigBuilder.getAuthConfig());
```

Comment:
If the kube config file is not available this will throw Exception, which we are ignoring.

Reply:
It gets validated in the post validator though

Reply:
Not in current implementation. We are directly type casting it. Even if we add validation there it would only be for null check. We won't have the correct error message in that case